src/components/dialogs/pick-request-type.vue

```diff
@@ -47,48 +47,49 @@
 </template>
 
 <script lang="ts">
-import newReqModule from '@/store/new-request-module'
 import { Component, Vue } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
+
+// Interfaces / enums / list data
 import { SelectOptionsI } from '@/interfaces'
+import { ActionBindingIF } from '@/interfaces/store-interfaces'
+import { RequestActions } from '@/store/list-data'
 
 @Component({})
 export default class PickRequestTypeDialog extends Vue {
-  get editMode () {
-    return newReqModule.editMode
-  }
+  // Global Getters
+  @Getter getPickRequestTypeModalVisible!: boolean
 
-  get showModal () {
-    return newReqModule.pickRequestTypeModalVisible
-  }
+  // Global Actions
+  @Action setClearErrors!: ActionBindingIF
+  @Action setExtendedRequestType!: ActionBindingIF
+  @Action setPickRequestTypeModalVisible!: ActionBindingIF
+  @Action setRequestAction!: ActionBindingIF
 
-  get location () {
-    return newReqModule.location
-  }
-
-  get requestActions () {
-    return newReqModule.requestActions
+  get showModal () {
+    return this.getPickRequestTypeModalVisible
   }
 
   set showModal (value: boolean) {
-    newReqModule.mutatePickRequestTypeModalVisible(value)
+    this.setPickRequestTypeModalVisible(value)
   }
 
   get tableData () {
-    let { length } = this.requestActions
+    let { length } = RequestActions
```

Comment:
This looks odd because the right hand side is PascalCase, which usually means it's a a type. I'd be tempted to rename it into camelCase.

In this case, I gather it's an array. But can it really be destructured this way to get array length???

Or I'd be explicit that I'm doing `const length = RequestActions.length`.